### PR TITLE
fix data race issue

### PR DESCRIPTION
**Pull request by williamaronli**
**Please provide a description of this PR:**
ref: https://github.com/istio/istio/issues/34115

the root cause is the go routinue blocked by the serve function and is executed after the tests completed. add a waitgroup to sync the goroutine
**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [X] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]

File: security/pkg/nodeagent/caclient/providers/citadel/client_test.go [hunk 1]
```diff
@@ -378,13 +379,18 @@ func TestCertExpired(t *testing.T) {
 		if err != nil {
 			t.Fatalf("failed to listen: %v", err)
 		}
+		var wg sync.WaitGroup
+		wg.Add(1)
 		go func() {
+			defer wg.Done()
 			pb.RegisterIstioCertificateServiceServer(s, &mockTokenCAServer{Certs: []string{}})
 			if err := s.Serve(lis); err != nil {
```
Comment (howardjohn) [inline security/pkg/nodeagent/caclient/providers/citadel/client_test.go:387]: s.Serve blocks, so the wg just hangs

**Review by howardjohn — CHANGES_REQUESTED** [review]

**Review by lei-tang — COMMENTED** [review]
Had a discussion with William offline. The waiting operation should wait for the grpc server stops and should be waiting after the test completes but before the test exits.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**